docs/topic/automatic-merging.rst

```diff
@@ -0,0 +1,61 @@
+.. _topic/automatic-merging:
+
+==========================
+Automatic Merging Behavior
+==========================
+
+``nbgitpuller`` tries to make sure the end user who clicked the link
+**never** has to manually interact with the git repo. This requires us to
+make some opinionated choices on how we handle various cases where both the
+student (end user) and instructor (author of the repo) repo have modified the
+repository.
+
+Here, how describe we handle the various possible cases each time the
+student clicks the nbgitpuller link.
+
+Case 1: Student and instructor changed different files
```

Comment:
```suggestion
Case 1: The instructor changes a file that the student has not changed
======================================================================
```

Reply:
(updated to keep tense consistent with other cases)